spec/avromatic/model/builder_spec.rb

```diff
@@ -1061,9 +1061,9 @@ def initialize
         end
 
         it "coerces a time to a union member" do
-          now = Time.now
-          instance = test_class.new(u: now)
-          expect(instance.u).to eq(Time.at(now.to_i, now.usec / 1000 * 1000))
+          time = Time.at(1540332724, 123456.789)
+          instance = test_class.new(u: time)
+          expect(instance.u).to eq(Time.at(1540332724, 123_000))
         end
```

Comment:
I think to test the bug fix you should use a value like `time = Time.at(1540332724, 123000.789)`. Then you have `time.usec % 1_000 == 0` but `time.nsec % 1_000_000 == 789`. That should fail to coerce with the old code but pass with your change.

With `time = Time.at(1540332724, 123456.789)`,  `time.usec % 1_000 == 456` so the old code would still coerce it.